### PR TITLE
[storage-proto] move TreeState and StartupInfo to storage-interface

### PR DESCRIPTION
**Pull request by lightmark**
## Motivation

storage-proto depends on `tonic` whereas most components in TCB needs `TreeState` and `StartupInfo`. As a result, they depend on `tonic` indirectly. 
After this PR, the dependency is removed. 

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Y

## Test Plan

CI

### CODE REVIEW

File: storage/storage-proto/src/tests.rs [hunk 1]
```diff
@@ -41,9 +43,6 @@ test_conversion!(
 test_conversion!(test_save_transactions_request, SaveTransactionsRequest);
 test_conversion!(test_get_transactions_request, GetTransactionsRequest);
 test_conversion!(test_get_transactions_response, GetTransactionsResponse);
-test_conversion!(test_tree_state, TreeState);
-test_conversion!(test_startup_info, StartupInfo);
-test_conversion!(test_get_startup_info_response, GetStartupInfoResponse);
```
Comment (msmouse) [inline storage/storage-proto/src/tests.rs:46]: no way to keep?

Comment (msmouse) [reply storage/storage-proto/src/tests.rs:46]: OIC we are deprecating GRPC.

@bors-libra delegate+

Comment (lightmark) [reply storage/storage-proto/src/tests.rs:46]: too much work. Not worth it.

**Comment by davidiw** [comment]
> @bors-libra r+

@bors-libra r=msmouse

or you just approved your own pr

**Comment by davidiw** [comment]
@bors-libra r+

free me!!!!

**Pull request closed without merging**